src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/factories.ts

```diff
@@ -583,6 +583,26 @@ export function getDateLiterals(options?: {
   ];
 }
 
+export function getControlSuggestionIfSupported(
+  supportsControls: boolean,
+  type: ESQLVariableType,
+  getVariablesByType?: (type: ESQLVariableType) => ESQLControlVariable[] | undefined
+) {
+  if (!supportsControls) {
+    return [];
+  }
+  if (supportsControls) {
+    const variableType = type;
+    const variables = getVariablesByType?.(variableType) ?? [];
+    const controlSuggestion = getControlSuggestion(
+      variableType,
+      variables?.map((v) => `?${v.key}`)
+    );
+    return controlSuggestion;
+  }
+  return [];
```

Comment:
The if and return statements seem redundant. Should this function always return control suggestions? Perhaps the supported check should be performed before invoking it.

```suggestion
  if (!supportsControls) {
    return [];
  }
    const variableType = type;
    const variables = getVariablesByType?.(variableType) ?? [];
    const controlSuggestion = getControlSuggestion(
      variableType,
      variables?.map((v) => `?${v.key}`)
    );
    return controlSuggestion;
```

Reply:
Good point, addressed here https://github.com/elastic/kibana/pull/210170/commits/7bb3bb14142baf4fbd9820d7feb558c6a292742a